src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -595,7 +376,7 @@ def cancel_job(self, job_id: str) -> bool:
             raise ValueError(f"No job present with id {job_id}")
 
         # otherwise, our job ID is fine
-        if job_id in self._completed_job_states:
+        if self.get_job(job_id).terminal_state:
```

Comment:
`get_job(job_id).is_finished()` might be more appropriate here and below. Maybe that's just stylistic and I'm being picky?